docs/DeveloperGuide.md

```diff
@@ -238,6 +238,49 @@ _{more aspects and alternatives to be added}_
 
 _{Explain here how the data archiving feature will be implemented}_
 
+### Find tag feature
+
+The find tag feature is used when a user is interested in finding contacts who have a certain `Tag`. Each `Person` has a set of `Tags` which contains unique `Tags` since each `Person` should not have more than 1 of the same tag.
+
+This feature is facilitated by `FindTagCommand`, which makes use of a `TagContainsKeywordPredicate` that checks if the `Tag` set of a `Person` contains all the tag names in the `List` of keywords (case-insensitive).
+
+The `FindTagCommand#execute()` method looks through the `Tag` set of each `Person` and updates the `Model#filteredPersons` using `Model#updateFilteredPersonsList()` which uses the `TagContainsKeywordPredicate`. The `Model` then displays the currently most updated filtered person list which reflects contacts with the specified tags in the list. 
+
+As such, `FindTagCommand` extends `Command` and executes the command by calling `FindTagCommand#execute()`.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If there are multiple keywords, e.g. `findtag friends colleagues`, then both contact should have both tags `friends` and `colleagues` in order for the contact to be reflected on the updated filtered person list.
+
+</div>
+
+Given below is an example usage scenario and how the find tag feature behaves at each step.
+
+Step 1. The user launches the address book for the first time, initialized with the initial address book state.
+
+Step 2. The user executes the command `findtag friends` to find contacts with the tag `friends` in their set of tags.
+
+Step 3. The AddressBookParser parses the command `findtag friends` and creates a `FindTagCommandParser` to parse the keyword `friends`.
+
+Step 4. The `TagContainsKeywordPredicate` is created and passed onto the constructor of `FindTagCommand` to create a new `FindTagCommand` object.
+
+Step 5. The `LogicManager` then calls `FindTagCommand#execute()`, calling `Model#updateFilteredPersonList()` which updates the list of persons to be displayed.
+
+Step 6. The `CommandResult` created from `FindTagCommand#execute()` is returned to the `LogicManager` which will be reflected in the `ResultDisplay` of the `GUI`.
+
+![FindTagSequenceDiagram](images/FindTagSequenceDiagram.png)
+
```

Comment:
I'm not sure if it's needed to constantly repeat that PlantUML can't handle terminating lifelines after the destroy marker (already mentioned in undo/redo part), do you think it would be good to just include it so that it's clear you are aware and can't do anything about it? :thinking:

Reply:
Yup that sounds great! I'll add it right now!